json-rpc/src/runtime.rs

```diff
@@ -113,30 +122,26 @@ async fn rpc_endpoint(
         .map_err(|_| reject::custom(DatabaseError))?;
 
     let resp = Ok(if let Value::Array(requests) = data {
-        if requests.len() > service.batch_size_limit as usize {
-            let data = ErrorData::ExceedBatchSizeLimit(ExceedBatchSizeLimit {
-                limit: service.batch_size_limit,
-                batch_request_size: requests.len(),
-            });
-            let mut resp = Map::new();
-            resp.insert(
-                "error".to_string(),
-                JsonRpcError::invalid_request_with_data(Some(data)).serialize(),
-            );
+        match service.validate_batch_size_limit(requests.len()) {
+            Ok(_) => {
+                // batch API call
+                let futures = requests.into_iter().map(|req| {
```

Comment:
I think we should be a bit more careful about concurrent write operations - if we have a batch request for multiple sequential txn submissions, the submissions might fail depending on what order they're executed